pkg/gadgettracermanager/containerutils/containerutils.go

```diff
@@ -156,45 +169,71 @@ func PidFromContainerId(containerID string) (int, error) {
 		}
 		return dockerInspect[0].State.Pid, nil
 	} else if strings.HasPrefix(containerID, "cri-o://") {
-		out, err := exec.Command("chroot", "/host", "crictl", "inspect", strings.TrimPrefix(containerID, "cri-o://")).Output()
+		IDWithoutPrefix := strings.TrimPrefix(containerID, "cri-o://")
+		r, err := getContainerStatus(CRIO_DEFAULT_SOCKET_PATH, IDWithoutPrefix)
 		if err != nil {
 			return -1, err
 		}
-		type CRIOInspect struct {
-			Pid int
+		pidStr, ok := r.Info["pid"]
+		if !ok {
+			return -1, fmt.Errorf("container status reply from runtime doesn't contain 'pid'")
 		}
-		var crioInspect CRIOInspect
-		err = json.Unmarshal(out, &crioInspect)
+
+		pid, err := strconv.Atoi(pidStr)
 		if err != nil {
 			return -1, err
 		}
-		if crioInspect.Pid == 0 {
-			return -1, fmt.Errorf("invalid pid")
-		}
-		return crioInspect.Pid, nil
+
+		return pid, nil
 	} else if strings.HasPrefix(containerID, "containerd://") {
-		out, err := exec.Command("chroot", "/host", "crictl", "inspect", strings.TrimPrefix(containerID, "containerd://")).Output()
+		IDWithoutPrefix := strings.TrimPrefix(containerID, "containerd://")
+		r, err := getContainerStatus(CONTAINERD_DEFAULT_SOCKET_PATH, IDWithoutPrefix)
 		if err != nil {
 			return -1, err
 		}
-		type ContainerdInspect struct {
-			Info struct {
-				Pid int
-			}
+		info, ok := r.Info["info"]
+		if !ok {
+			return -1, fmt.Errorf("container status reply from runtime doesn't contain 'info'")
 		}
-		var containerdInspect ContainerdInspect
-		err = json.Unmarshal(out, &containerdInspect)
-		if err != nil {
+
+		containerdInspect := struct{ Pid int }{}
+		if err := json.Unmarshal([]byte(info), &containerdInspect); err != nil {
 			return -1, err
 		}
-		if containerdInspect.Info.Pid == 0 {
+
+		if containerdInspect.Pid == 0 {
```

Comment:
Should we do this check for cri-o as well? Or remove it here?

Reply:
I removed the check, if the parsing succeeds then the PID should be good.